client/client.go

```diff
@@ -240,19 +240,25 @@ func WithMaxErrorRetry(count int) ClientOption {
 var _ Client = (*client)(nil)
 
 type client struct {
-	keyspaceID uint32
-	// svcDiscovery is for pd service discovery
-	svcDiscovery    ServiceDiscovery
-	tsoClient       *tsoClient
+	keyspaceID      uint32
+	svrUrls         []string
+	pdSvcDiscovery  ServiceDiscovery
 	tokenDispatcher *tokenDispatcher
 
+	// For service mode switching
+	mutex           sync.RWMutex
```

Comment:
How about declaring an anonymous struct to tell which fields this lock is protecting?

```suggestion
	struct {
		mutex           sync.RWMutex
		xxx
	}
```

Reply:
```
// serviceModeKeeper is for service mode switching
type serviceModeKeeper struct {
	svcModeMutex    sync.RWMutex
	serviceMode     pdpb.ServiceMode
	tsoClient       atomic.Value
	tsoSvcDiscovery ServiceDiscovery
}

type client struct {
        ...
	serviceModeKeeper
        ...
```